### DISCUSSION TITLE
Why there is no velocity commands? we have to use only the pose (positional) data

### DISCUSSION THREAD

**Discussion by hinnatariq** [larics/icuas22_competition#62] [Q&A]
I have a question @fpetric, 

Seeing the topics available for the UAV, there is no velocity commands, we only have to use pose (positional) data. Is it a limitation for this compeition? as mostly velocity topic are available to use for working on UAVs. I know its late to ask, but I am still curious as its causing issues to control the red uav properly, especially the yaw seems difficult to control.

**Comment by fpetric** [comment]
Hi, 
you can control velocities to a degree using feedforward gains (see readme and discussion [here](https://github.com/larics/icuas22_competition/discussions/40)). The reason behind the controller structure is a complex one, but it is what we use on the real platform for the finals and we wanted the simulation to be as similar as possible. If you want to control velocities, you were allowed to implement your own controller for the simulation phase (see [here](https://github.com/larics/icuas22_competition/discussions/34#discussioncomment-2360114)), but we cannot allow custom controllers for the finals as we will not have enough time in Dubrovnik to check and debug everything.
